packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -169,30 +211,193 @@ class IOSDeploy {
     return true;
   }
 
-  // Maps stdout line stream. Must return original line.
-  String _monitorFailure(String stdout) {
-    // Installation issues.
-    if (stdout.contains(noProvisioningProfileErrorOne) || stdout.contains(noProvisioningProfileErrorTwo)) {
-      _logger.printError(noProvisioningProfileInstruction, emphasis: true);
+  String _monitorFailure(String stdout) => _monitorIOSDeployFailure(stdout, _logger);
+}
+
+/// Wrapper to launch app and attach the debugger with ios-deploy.
+class IOSDeployDebugger {
+  IOSDeployDebugger({
+    @required Logger logger,
+    @required ProcessUtils processUtils,
+    @required List<String> launchCommand,
+    @required Map<String, String> iosDeployEnv,
+    StreamController<String> debuggerOutput,
+  }) : _processUtils = processUtils,
+        _logger = logger,
+        _launchCommand = launchCommand,
+        _iosDeployEnv = iosDeployEnv,
+        _debuggerOutput = debuggerOutput ?? StreamController<String>.broadcast();
+
+  factory IOSDeployDebugger.test({
```

Comment:
nit: maybe describe what parts are hard-coded for testing

Reply:
Done.